src/app_client.rs

```diff
@@ -51,10 +56,55 @@ async fn process_block(
 				rpc_fetched.len()
 			);
 
-			let unfetched_count = positions.len() - ipfs_fetched.len() - rpc_fetched.len();
-			if unfetched_count > 0 {
-				return Err(anyhow!("Failed to fetch {} cells", unfetched_count));
+			let mut cells = vec![];
+			cells.extend(ipfs_fetched);
+			cells.extend(rpc_fetched);
+
+			if positions.len() > cells.len() {
+				return Err(anyhow!(
+					"Failed to fetch {} cells",
+					positions.len() - cells.len()
+				));
+			}
+
+			let count = verify_proof(
```

Comment:
@prabal-banerjee for cells belonging to the same appID (sequentially ordered), does it make sense at all to verify proof, or should the app client just check the block confidence and then proceed with the reconstruction?

Reply:
on the other hand, it does make sense to verify the proof, but the sequential cell proof verification mustn't be used for confidence score. So we should be good here...

Reply:
IMO, the app clients should verify all downloaded cells. The confidence score should be just computed on the LC and not the app client.